drivers/ethernet/dsa_ksz8794.c

```diff
@@ -961,10 +957,33 @@ static void dsa_iface_init(struct net_if *iface)
 				     NET_LINK_ETHERNET);
 		ctx->dsa_port_idx = i;
 		ctx->dsa_ctx = context;
+
+		/*
+		 * Initialize ethernet context 'work' for this iface to
+		 * monitor the status of it (after SHA1: c1f7b9f45a78d7e082a1):
+		 *
+		 * ctx->iface = iface;
+		 * k_work_init(&ctx->carrier_work, carrier_on_off);
+		 *
+		 * As now carrier_work is used to call carrier_on_off(), which
+		 * in turn calls ethernet_mgnt* functions.
+		 */
+		ethernet_init(iface);
```

Comment:
I think this is a too detailed comment and it will quickly get outdated (for example because of the 2 C code lines). In fact the most important thing to do is to call `ethernet_init(iface)` before calling `net_eth_carrier_on(iface)` and `net_eth_carrier_on(iface)`. I don't see why `c1f7b9f45a78d7e082a1` has to be referenced in this comment.